### PR TITLE
DS-2964 Remove NETs during stacking

### PR DESCRIPTION
**Pull request by jrwishart**
The code to identify and flag columns for removal during data stacking
has been updated. It will now identify and remove columns from the data
that consist of NETs that are not entirely comprised of unique variables
that are not observed elsewhere in the data. That is, the NET will be
removed if any variable in its composition is observed elsewhere. This
is guided by the information in the (secondary)codeframe as necessary.

### CODE REVIEW

**Comment by jrwishart** [comment]
Hold off on the merge, still need to add a warning to the user when the codeframe shows a NET that includes a unique element not seen in the other codeframe elements. (i.e. something has been hidden and cannot be recovered and added back into the data.frame corresponding to the grid variable set.)

File: R/regression.R [hunk 1]
```diff
@@ -1912,6 +1960,35 @@ updateStackedFormula <- function(data, formula)
...
+    variable.type <- reduction.list[["variable.type"]]
+    net.name.txt <- paste0(sQuote(nets, q = FALSE))
+    if (length(nets) > 1)
+        net.name.txt <- paste0(": (", paste0(net.name.txt, collapse = "; "), ")")
+    net.txt <- ngettext(length(nets), "a NET, ", "NETs")
+    variable.type.and <- paste0(paste0(variable.type, collapse = " and "), " variables")
+    variable.type.or <- paste0(paste0(variable.type, collapse = " or "), " variables")
+    net.txt.2 <- ngettext(length(nets), "this NET and its hidden", "these NETs and their hidden")
+    warning("NETs are removed from this analysis unless all their codes are not observed elsewhere. ",
+            "The ", variable.type.and, " have ", net.txt, net.name.txt, ", with both hidden and observed codes. ",
```
Comment (mwmclean) [inline R/regression.R:1986]: where is the "both hidden and observed codes" portion of this warning coming from?

**Review by mwmclean — APPROVED** [review]

**Comment by mwmclean** [comment]
Please add a sentence or two mentioning NET removal in discussion of stacking in the `Regression()` help file. The big discussion already in the {{unstacked.data}} parameter section should be moved to Details.

File: R/regression.R (hunk 1 above)
Comment (jrwishart) [reply R/regression.R:1986]: Updated/rephrased in https://github.com/Displayr/flipRegression/commit/efac6e9578e3e4e9fda3f18d26bfdb84ce13779c.

**Pull request merged**